### PR TITLE
Fix for transient issues, Fix for the requests encountering device_network_not_available

### PR DESCRIPTION
**Pull request by tanmaymanolkar1**
Created a new category TRANSIENT to help categorize the issues, which can help in not caching them if needed and taking the necessary steps to service these failures

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java [hunk 1]
```diff
@@ -512,7 +514,9 @@ private static boolean eligibleToCache(CommandResult commandResult) {
      * @return
      */
     private static boolean eligibleToCacheException(BaseException exception) {
-        if (exception instanceof IntuneAppProtectionPolicyRequiredException) {
+        if (exception instanceof IntuneAppProtectionPolicyRequiredException
+                || (exception instanceof ClientException
+                && (ErrorStrings.DEVICE_NETWORK_NOT_AVAILABLE.equals(((ClientException)exception).getErrorCode())))) {
```
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java:519]:
```suggestion
                || ErrorStrings.DEVICE_NETWORK_NOT_AVAILABLE.equals(((ClientException)exception).getErrorCode())) {
```

**Review by AdamBJohnsonx — APPROVED** [review]
We'll want a follow-on task to capture the entire class of these.

File: common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java (hunk 1 above)
Comment (iambmelt) [reply common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java]: +1 on this feedback

Comment (AdamBJohnsonx) [reply common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java]: > +1 on this feedback

And the fact that it's ClientException is largely my fault, TTYTT.

**Review by iambmelt — APPROVED** [review]
Approving with comments (do not cast)

**Review by shoatman — APPROVED** [review]

File: common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java [hunk 2]
```diff
@@ -512,7 +514,9 @@ private static boolean eligibleToCache(CommandResult commandResult) {
      * @return
      */
     private static boolean eligibleToCacheException(BaseException exception) {
-        if (exception instanceof IntuneAppProtectionPolicyRequiredException) {
+        //TODO : ADO 1373343 Add the whole transient exception category.
+        if (exception instanceof IntuneAppProtectionPolicyRequiredException
+                || ErrorStrings.DEVICE_NETWORK_NOT_AVAILABLE.equals(((ClientException)exception).getErrorCode())) {
```
Comment (shahzaibj) [inline common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java:519]: you probably want to check that it is instance of `ClientException` before you cast

Comment (iambmelt) [reply common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java:519]: (No cast is required here)

Comment (shahzaibj) [reply common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java:519]: Looks like you have already got that feedback

Comment (tanmaymanolkar1) [reply common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java:519]: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1329#discussion_r619458892

**Pull request merged**